wire-ios-data-model/Source/ManagedObjectContext/Migration/2.110-2.111/RemoveDuplicatePreAction.swift

```diff
@@ -29,10 +29,26 @@ class RemoveDuplicatePreAction: CoreDataMigrationAction {
 
     override func execute(in context: NSManagedObjectContext) {
         entityNames.forEach { entityName in
+            removeNilPrimaryKey(for: entityName, context: context)
             removeDuplicates(for: entityName, context: context)
         }
     }
 
+    // Method to cleanup objects without remoteIdentifierDataKey before removeDuplicates
+    private func removeNilPrimaryKey(for entityName: String, context: NSManagedObjectContext) {
+        let request = NSFetchRequest<NSManagedObject>(entityName: entityName)
+        request.predicate = NSPredicate(format: "%K = nil", ZMManagedObject.remoteIdentifierDataKey())
+        do {
+            let objectsWithNil = try context.fetch(request)
+
+            for object in objectsWithNil {
+                context.delete(object)
+            }
```

Comment:
Maybe we can add a log how many objects will be deleted to make this step visible in the happy case.

Reply:
done